src/api/callback.cc

```diff
@@ -64,8 +64,15 @@ InternalCallbackScope::InternalCallbackScope(Environment* env,
   Isolate* isolate = env->isolate();
 
   HandleScope handle_scope(isolate);
-  // If you hit this assertion, you forgot to enter the v8::Context first.
-  CHECK_EQ(Environment::GetCurrent(isolate), env);
+  Local<Context> current_context = isolate->GetCurrentContext();
+  // If you hit this assertion, the caller forgot to enter the right Node.js
+  // Environment's v8::Context first.
+  // We first check `env->context() != current_context` because the contexts
+  // likely *are* the same, in which case we can skip the slightly more
+  // expensive Environment::GetCurrent() call.
+  if (UNLIKELY(env->context() != current_context)) {
+    CHECK_EQ(Environment::GetCurrent(isolate), env);
+  }
 
   env->isolate()->SetIdle(false);
```

Comment:
```suggestion
  isolate->SetIdle(false);
```
Just in case you touch the file again during this PR, otherwise feel free to ignore.